woocommerce/woocommerce-android
# Domain change: Loading of site domains & active plan

Fixes #8278, #8280, #8281, subtasks of #8238.

The PR adds an implementation for loading of the assigned free site domain and paid domains for a site. Also, the current site plan is loaded in order to determine if a site has a free domain credit, which is used show/hide domain claim banner.

**To test:**
1. Login and load a site that has paid domains
2. Go to Settings -> Domains
3. After loading, verify the correct free & paid domains are listed and the correct primary domain is marked
4. Go back to a site picker
5. Pick a site that has a plan purchased and a free domain credit
6. Go to Settings -> Domains
7. Verify the domain claim banner is displayed
